src/components/channels/ChannelsSubscriber.js

```diff
@@ -23,99 +23,117 @@ const VIEWER_PALETTE = [
-    PubSub.publish(COLORS_CHANGE, { channel, rgb });
-  }
 
-  toggleChannel(channel) {
-    this.setState((prevState) => {
-      const channelToggle = !prevState.channelsOn[channel];
-      const channelOn = { [channel]: channelToggle };
-      PubSub.publish(CHANNEL_TOGGLE, { channel, channelToggle });
-      return { channelsOn: { ...prevState.channelsOn, ...channelOn } };
+      setChannelNames(cNames);
+      setSliderDomains(domains.map(d => (Array.isArray(d) ? d : [0, STANDARD_MAX])));
+      setEventId(id);
+    }
+    memoizedOnReady();
+    const token = PubSub.subscribe(RASTER_ADD, handleRasterAdd);
+    return () => PubSub.unsubscribe(token);
+  }, [memoizedOnReady]);
+
+  const handleColorChange = (i, rgb) => {
+    setColorValues((prevColorValues) => {
+      const nextColorValues = [...prevColorValues];
+      nextColorValues[i] = rgb;
+      PubSub.publish(COLORS_CHANGE + eventId, nextColorValues);
+      return nextColorValues;
     });
-  }
+  };
+
+  const handleSliderChange = (i, sliderValue) => {
+    setSliderValues((prevSliderValues) => {
+      const nextSliderValues = [...prevSliderValues];
+      nextSliderValues[i] = sliderValue;
+      PubSub.publish(SLIDERS_CHANGE + eventId, nextSliderValues);
+      return nextSliderValues;
+    });
+  };
+
+  const handleChannelIsOnChange = (i) => {
+    setChannelIsOn((prevChannelIsOn) => {
+      const nextChannelIsOn = [...prevChannelIsOn];
+      nextChannelIsOn[i] = !prevChannelIsOn[i];
+      PubSub.publish(CHANNEL_TOGGLE + eventId, nextChannelIsOn);
+      return nextChannelIsOn;
+    });
+  };
 
-  render() {
-    const { colorValues, channelsOn, rangeValues } = this.state;
-    const hr = <hr style={{ border: '1px solid #000' }} />;
-    const channelSliders = Object.keys(colorValues)
-      .map((channel, i) => {
-        const channelColor = colorValues[channel] || VIEWER_PALETTE[i];
-        const rangeValue = rangeValues[channel] || [0, STANDARD_MAX];
-        return (
-          <div key={`container-${channel}`}>
-            <div>{channel}</div>
-            <div className="channel-container">
-              <Checkbox
-                className="channel-checked"
-                checked={channelsOn[channel]}
-                onChange={() => this.toggleChannel(channel)}
-              />
-              <PopoverColor
-                prefixClass="channel"
-                color={channelColor}
-                setColor={rgb => this.setColorValue({ channel, rgb })}
-                placement="left"
-                palette={VIEWER_PALETTE}
-              />
-              <ChannelSlider
-                channel={channel}
-                setSliderValue={this.setSliderValue}
-                range={rangeValue}
-                color={channelColor}
-              />
-            </div>
+  if (channelNames && colorValues && sliderValues && channelIsOn && sliderDomains) {
+    const channelSliders = channelNames.map((name, i) => {
+      const checked = channelIsOn[i];
```

Comment:
the other names are consistent between the array and the element... not sure if there's a nice way to do that for this one.

Reply:
Yeah I'm trying to think if there is a better name for `channelIsOn`, but this is the name used by viv. The issue is that `channelIsOn` is not pural like the others, so it doesn't appropriately describe what the variable is, which is a boolean array indicating which channels are on and which are off. Would you have any suggestions for a better name?

Reply:
I'm open to other suggestions for `viv`.  I thought `channelIsToggled` or something involving "toggle" wasn't good.  I agree this is confusing.

Reply:
Maybe `channelVisibilities` and `setChannelVisibilities`?